row/fetcher: clean up and harden TestRowFetcherMVCCMetadata

yuzefovich opened the pull request:
This commit refactors `TestRowFetcherMVCCMetadata` a bit to explicitly request `crdb_internal_mvcc_timestamp` as opposed to fishing it out of the fetcher, which allows removing an exported method from the latter.

This commit also hardens the test to de-flake it when write buffering is enabled. The test scans the storage engine directly, so we need to tweak a testing knob to ensure that COMMIT blocks until the writes are committed into pebble.

Fixes: #147174.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/147182)
<!-- Reviewable:end -->

yuzefovich committed:
row/fetcher: clean up and harden TestRowFetcherMVCCMetadata

This commit refactors `TestRowFetcherMVCCMetadata` a bit to explicitly
request `crdb_internal_mvcc_timestamp` as opposed to fishing it out of
the fetcher, which allows removing an exported method from the latter.

This commit also hardens the test to de-flake it when write buffering is
enabled. The test scans the storage engine directly, so we need to tweak
a testing knob to ensure that COMMIT blocks until the writes are
committed into pebble.

Release note: None

michae2 reviewed (approved):
Nice! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (But I defer to @stevendanna.)

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147182#-:-OQyFyGi8xb11FHMjKvJ:b-9okzow)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @stevendanna)



<!-- Sent from Reviewable.io -->
